crypto/store/store_init.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2001-2016 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/err.h>
+#include "internal/store.h"
+#include "store_local.h"
+
+static CRYPTO_ONCE store_init = CRYPTO_ONCE_STATIC_INIT;
+DEFINE_RUN_ONCE_STATIC(do_store_init)
+{
+    return OPENSSL_init_crypto(0, NULL)
```

Comment:
Have you actually tried that simple program without the OPENSSL_init_crypto() call? AFAICT, if you don't call it, then as soon as anything goes near the err library, ERR_get_state() gets called internally, which initialises itself. If you've tried it and it doesn't do that, I'm not sure why.